### PR TITLE
Make che-machine-exec track activity and stop workspace CR by idle timeout

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
This PR makes che-machine-exec track activity and stop workspace CR by idle timeout.

### Which issues does it solves?
It solves https://github.com/eclipse/che/issues/16683

### CODE REVIEW

File: cfg/cfg.go [hunk 1]
```diff
@@ -102,4 +114,8 @@ func Print() {
 	if UseBearerToken {
 		logrus.Infof("==> Authenticated user ID: %s", AuthenticatedUserID)
 	}
+	if IdleTimeout > 0 {
```
Comment (JPinkney) [inline cfg/cfg.go:117]: Do we also need to check stop-retry-period as well to make sure it's not negative?

File: activity/manager.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+type Manager interface {
+	//Start starts users activity and scheduling workspace stoppping if there is no activity for idle timeout
+	Start()
+
+	//Tick registers users activity and postpones workpace stopping by inactivity
+	Tick()
+}
+
+func New(idleTimeout, stopRetryPeriod time.Duration) (Manager, error) {
+	if idleTimeout < 0 {
```
Comment (JPinkney) [inline activity/manager.go:52]: Should we check if stopRetryPeriod is negative here as well?

File: cfg/cfg.go (hunk 1 above)
Comment (sleshchenko) [reply cfg/cfg.go:117]: I suppose stop-retry-period must not be negative. Will add validation.

File: activity/manager.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
+//
+// Copyright (c) 2012-2019 Red Hat, Inc.
```
Comment (AndrienkoAleksandr) [inline activity/manager.go:2]: Minor 2020

File: activity/manager.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+			logrus.Info("Workspace is successfully stopped by inactivity. Bye")
+			return
+		}
+	case <-m.activityC:
+		logrus.Debug("Activity is reported. Resetting timer")
+		if !timer.Stop() {
+			<-timer.C
+		}
+		timer.Reset(m.idleTimeout)
+	}
```
Comment (amisevsk) [inline activity/manager.go:122]: This needs to be done in a goroutine, otherwise it blocks startup.

Comment (sleshchenko) [reply activity/manager.go:122]: Makes sense. I wonder why it worked in my local testing :exploding_head:

**Comment by sleshchenko** [comment]
Now it seems to work as expected. Thank Angle for testing and finding critical bug!
I'm planning to try return pointer func to make sure that the activity manager is not run twice and even maybe move running go routing on manager.Start level, as it's done with server.Run().

**Pull request merged**